### PR TITLE
Extract logic from output document template

### PR DESCRIPTION
**Pull request by stevenwilkin**
The logic now resides with the the domain object, away from the template. CSV generation should now be relatively straightforward.

FYI, there is a pending spec for `OutputDocument#greeting`.

### CODE REVIEW

**Comment by gjvis** [comment]
It is a thing of beauty. Nice. I think this is a much stronger place to extend from...

epichighfive

File: app/domain/output_document.rb [hunk 1]
```diff
@@ -3,10 +3,84 @@ class OutputDocument
...
+  delegate :address_line_1, :address_line_2, :address_line_3, :town, :county,
+           :postcode,
+           to: :appointment_summary, prefix: :attendee
+
   def initialize(appointment_summary)
     @appointment_summary = appointment_summary
   end
 
+  def attendee_name
+    "#{appointment_summary.title} #{appointment_summary.first_name} #{appointment_summary.last_name}"
```
Comment (andrewgarner) [inline app/domain/output_document.rb]: First name is optional, so would end up with double space like `Mr  Garner`. There is also a chance the first name and last name attributes could have whitespace as I don't believe they are sanitised before write. However, Rails has your back!

```rb
"#{appointment_summary.title} #{appointment_summary.first_name} #{appointment_summary.last_name}".squish
```

File: app/templates/output_document_introduction.html.erb [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      <% if income_in_retirement == 'pension' %>
+        <p>How much tax you’ll pay - the tax-free Personal Allowance for the
+          2015/16 tax year is £10,600. </p>
+
+        <p>How long your money needs to last - on average, people aged 55
+          today will live to their mid-to-late 80s. </p>
+
+        <p>Future costs in retirement - you may have extra health and care
+          costs as you get older.
+          The impact of inflation - as prices go up your pension might buy you
```
Comment (gjvis) [inline app/templates/output_document_introduction.html.erb]: Based on the design I think this should be two separate paragraphs.

File: app/domain/output_document.rb (hunk 1 above)
Comment (stevenwilkin) [inline app/domain/output_document.rb]: I didn't know about `String#squish`, nice!

File: app/templates/output_document_introduction.html.erb (hunk 2 above)
Comment (stevenwilkin) [inline app/templates/output_document_introduction.html.erb]: Sorted

File: app/domain/output_document.rb [hunk 3]
```diff
@@ -3,10 +3,84 @@ class OutputDocument
...
+
+  def guider_organisation
+    if appointment_summary.guider_organisation == 'tpas'
+      'The Pensions Advisory Service'
+    else
+      'Pension Wise'
+    end
+  end
+
+  def value_of_pension_pots
```
Comment (andrewgarner) [inline app/domain/output_document.rb]: This needs updating to take into consideration whether the value given is an approximation.

If `value_of_pension_pots_is_approximate == true` then the output should read `£50,000 (approximately)`. If `!upper_value_of_pension_pots.blank?`, then ignore `value_of_pension_pots_is_approximate`.

File: app/domain/output_document.rb [hunk 4]
```diff
@@ -3,10 +3,84 @@ class OutputDocument
...
+        'tailored'
+      else
+        'generic'
+      end
+    else
+      'other'
+    end
+  end
+
+  def greeting
```
Comment (andrewgarner) [inline app/domain/output_document.rb]:
```rb
"You recently had a Pension Wise guidance appointment with #{guider_name} from #{guider_organisation} on #{appointment_date}"
```

File: app/domain/output_document.rb [hunk 5]
```diff
@@ -3,10 +3,84 @@ class OutputDocument
...
+  end
+
+  def appointment_reference
+    appointment_summary.reference_number
```
Comment (andrewgarner) [inline app/domain/output_document.rb]: There is a chance we could have multiple summaries for the same appointment. To cover that off I used the following reference in the documents issued to date:

```rb
"#{appointment_summary.id}/#{appointment_summary.reference_number}" # 54/12344
```

File: app/domain/output_document.rb [hunk 6]
```diff
@@ -3,10 +3,84 @@ class OutputDocument
...
   def initialize(appointment_summary)
     @appointment_summary = appointment_summary
   end
 
+  def attendee_name
+    "#{appointment_summary.title} #{appointment_summary.first_name} #{appointment_summary.last_name}".squish
+  end
+
+  def appointment_date
+    appointment_summary.date_of_appointment.to_s(:long)
```
Comment (andrewgarner) [inline app/domain/output_document.rb]: Does `(:long)` == https://www.gov.uk/service-manual/user-centred-design/resources/patterns/dates.html#formatting-dates?

File: app/domain/output_document.rb (hunk 4 above)
Comment (gjvis) [inline app/domain/output_document.rb]: https://github.com/guidance-guarantee-programme/output/issues/79

File: app/domain/output_document.rb (hunk 5 above)
Comment (gjvis) [inline app/domain/output_document.rb]: https://github.com/guidance-guarantee-programme/output/issues/80

File: app/domain/output_document.rb (hunk 3 above)
Comment (gjvis) [inline app/domain/output_document.rb]: In progress currently in another PR

File: app/domain/output_document.rb (hunk 6 above)
Comment (gjvis) [inline app/domain/output_document.rb]: https://github.com/guidance-guarantee-programme/output/issues/81

**Comment by gjvis** [comment]
Comments extracted into issues to be solved as separate PRs. I will rebase and merge this PR shortly.

:rocket:

**Pull request merged**